model/push_event_context.go

```diff
@@ -0,0 +1,93 @@
+package model
+
+import (
+	"strings"
+
+	"github.com/google/go-github/v45/github"
+	"github.com/mattermost/release-bot/log"
+)
+
+type PushEventContext struct {
+	event          string
+	repository     string
+	reference      string
+	tag            bool
+	fork           bool
+	name           string
+	commitHash     string
+	installationID int64
+}
+
+func newPushEventContext(event github.PushEvent) EventContext {
+	isTag := strings.HasPrefix(event.GetRef(), "refs/tags/")
+	var name string
+	if isTag {
+		name = strings.TrimPrefix(event.GetRef(), "refs/tags/")
+	} else {
+		name = strings.TrimPrefix(event.GetRef(), "refs/heads/")
+	}
+	return &PushEventContext{
+		event:          "push",
+		repository:     event.GetRepo().GetFullName(),
+		reference:      event.GetRef(),
+		tag:            isTag,
+		fork:           event.GetRepo().GetFork(),
+		name:           name,
+		commitHash:     event.GetAfter(),
+		installationID: event.GetInstallation().GetID(),
+	}
+}
+
+func (pec *PushEventContext) Log() {
+	log.Info(pec.event).
+		Str("repo", pec.repository).
+		Str("ref", pec.reference).
+		Str("name", pec.name).
+		Bool("tag", pec.tag).
+		Bool("fork", pec.fork).
+		Str("hash", pec.commitHash).
+		Msg("")
+}
+
+func (pec *PushEventContext) GetEvent() string {
+	return pec.event
+}
+func (pec *PushEventContext) IsFork() bool {
+	return pec.fork
+}
+func (pec *PushEventContext) IsPr() bool {
+	return false
+}
+func (pec *PushEventContext) IsBranch() bool {
+	return !pec.tag
+}
+func (pec *PushEventContext) IsTag() bool {
+	return pec.tag
+}
+func (pec *PushEventContext) GetWorkflow() string {
+	return ""
```

Comment:
should be empty? same for other functions here

Reply:
For the push events there is no workflow identifier.

Reply:
And push events will be tested at follow-up PR.

Reply:
Added comment to those, for push-events there is no workflow definition. Github will send another event if workflow is started for that commit.